src/modules/email-publisher/EmailService.ts

```diff
@@ -0,0 +1,50 @@
+import * as util from 'util'
+import * as crypto from 'crypto'
+import * as kafka from 'kafka-node'
+
+import { Email } from './Email'
+import { Message, type } from './message/Message'
+
+export class EmailService {
+    private readonly producer: kafka.Producer;
+
+    constructor (private applicationIdentifier: string, kafkaClient: kafka.KafkaClient) {
+        if (applicationIdentifier == null) {
+            throw new Error('Application identifier is required')
+        }
+        if (kafkaClient == null) {
+            throw new Error('Kafka client is required')
+        }
+        this.producer = new kafka.Producer(kafkaClient)
+    }
+
+    public async send(email: Email): Promise<void> {
+        if (email == null) {
+            throw new Error('Email must be defined');
+        }
+        const message = this.createMessageFrom(email);
+        return this.sendMessage(message);
+    }
+
+    private async sendMessage(message: Message): Promise<void> {
+        const send = util.promisify(this.producer.send).bind(this.producer)
+        return send([{
+            topic: 'email-send',
```

Comment:
should we make this topic configurable?

Reply:
It won't change likely I assume so I decided to keep it hardcoded.